### PR TITLE
resolve: derive diag for undetermined macro resolution

### PR DESCRIPTION
**Pull request by bvanjoi**
simply for neatness.

### CODE REVIEW

File: compiler/rustc_resolve/src/macros.rs [hunk 1]
```diff
@@ -719,13 +720,14 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
...
-                        kind.descr(),
-                        Segment::names_to_string(path)
-                    );
-                    let msg_note = "import resolution is stuck, try simplifying macro imports";
-                    this.tcx.sess.struct_span_err(span, msg).note(msg_note).emit();
+                    this.tcx
+                        .sess
+                        .create_err(CannotDetermineMacroResolution {
+                            span,
+                            kind: kind.descr().to_string(),
```
Comment (cjgillot) [inline compiler/rustc_resolve/src/macros.rs:727]: Make `kind` a `&'static str`? Or impl `DiagnosticArg` for `MacroKind` directly?

**Review by cjgillot — COMMENTED** [review]
one nit and r=me

File: compiler/rustc_resolve/src/macros.rs [hunk 2]
```diff
@@ -719,13 +720,14 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
...
-                    let msg_note = "import resolution is stuck, try simplifying macro imports";
-                    this.tcx.sess.struct_span_err(span, msg).note(msg_note).emit();
+                    this.tcx
+                        .sess
+                        .create_err(CannotDetermineMacroResolution {
+                            span,
+                            kind: kind.descr(),
+                            path: Segment::names_to_string(path),
+                        })
+                        .emit();
```
Comment (compiler-errors) [inline compiler/rustc_resolve/src/macros.rs:730]:
```suggestion
                        .emit_err(CannotDetermineMacroResolution {
                            span,
                            kind: kind.descr(),
                            path: Segment::names_to_string(path),
                        });
```

**Comment by compiler-errors** [comment]
cjgillot said r=them after the nit, so

@bors r=cjgillot

**Pull request merged**